lib/std/time.zig

```diff
@@ -219,3 +224,1121 @@ test "Timer" {
+    pub fn weekdayName(self: *const Date) []const u8 {
+        return @tagName(@intToEnum(Weekday, self.dayOfWeek()));
+    }
+
+    // Return the name of the day of the month, eg "January"
+    pub fn monthName(self: *const Date) []const u8 {
+        assert(self.month >= 1 and self.month <= 12);
+        return @tagName(@intToEnum(Month, self.month));
+    }
+
+    // ------------------------------------------------------------------------
+    // Operations
+    // ------------------------------------------------------------------------
+
+    // Return a copy of the date shifted by the given number of days
+    pub fn shiftDays(self: *const Date, days: i32) Date {
+        return self.shift(Delta{.days=days});
+    }
+
+    // Return a copy of the date shifted by the given number of years
+    pub fn shiftYears(self: *const Date, years: i16) Date {
+        return self.shift(Delta{.years=years});
+    }
+
+    pub const Delta = struct {
+        years: i16 = 0,
+        days: i32 = 0,
+    };
+
+    // Return a copy of the date shifted in time by the delta
+    pub fn shift(self: *const Date, delta: Delta) Date {
+        if (delta.years == 0 and delta.days == 0) {
+            return self.copy() catch unreachable;
+        }
+
+        // Shift year
+        var year = self.year;
+        if (delta.years < 0) {
+            year -= @intCast(u16, -delta.years);
+        } else {
+            year += @intCast(u16, delta.years);
+        }
+        var ord = daysBeforeYear(year);
+        var days = self.dayOfYear();
+        const fromLeap = isLeapYear(self.year);
+        const toLeap = isLeapYear(year);
+        if (days == 59 and fromLeap and toLeap) {
+            // No change before leap day
+        } else if (days < 59) {
+            // No change when jumping from leap day to leap day
+        } else if (toLeap and !fromLeap) {
+            // When jumping to a leap year to non-leap year
+            // we have to add a leap day to the day of year
+            days += 1;
+        } else if (fromLeap and !toLeap) {
+            // When jumping from leap year to non-leap year we have to undo
+            // the leap day added to the day of yearear
+            days -= 1;
+        }
+        ord += days;
+
+        // Shift days
+        if (delta.days < 0) {
+            ord -= @intCast(u32, -delta.days);
+        } else {
+            ord += @intCast(u32, delta.days);
+        }
+        return Date.fromOrdinal(ord);
+    }
+
+};
+
+
+test "date-now" {
+    var date = Date.now();
+}
+
+test "date-compare" {
+    var d1 = try Date.create(2019, 7, 3);
+    var d2 = try Date.create(2019, 7, 3);
+    var d3 = try Date.create(2019, 6, 3);
+    var d4 = try Date.create(2020, 7, 3);
+    assert(d1.eql(&d2));
+    assert(d1.gt(&d3));
+    assert(d3.lt(&d2));
+    assert(d4.gt(&d2));
+}
+
+test "date-from-ordinal" {
+    var date = Date.fromOrdinal(9921);
+    testing.expect(date.year == 28);
```

Comment:
Use `expectEqual`. Could also compare the whole struct rather than the fields individually